oauthenticator/github.py

```diff
@@ -43,23 +47,80 @@ class GitHubMixin(OAuth2Mixin):
 
 
 class GitHubLoginHandler(OAuthLoginHandler, GitHubMixin):
-    pass
+    """We check the environment variables GITHUB_USE_ORGANIZATIONS,
+    GITHUB_USE_PUSH_TOKEN, GITHUB_USE_PRIVATE_PUSH_TOKEN, and
+    GITHUB_USE_EMAIL in order to set up the scope for the token we
+    request.
+
+    Each of these turns on a feature if it is set.
+
+    GITHUB_USE_ORGANIZATIONS enables the use of GitHub organizations
+    to allow provisioning of backend gids, which requires "read:org"
+    scope to iterate through the user's organizations and map their
+    names to their id numbers.
+
+    GITHUB_USE_PUSH_TOKEN requests "public_repo" access in order to
+    push code into public repositories--we use magic on the backend to
+    cache the GitHub token and set up .git-credentials with it.
+
+    GITHUB_USE_PRIVATE_PUSH_TOKEN does the same but with "repo"
+    access, so it can access both public and private repositories.
+
+    GITHUB_USE_EMAIL looks at the GitHub email field; this is used to
+    set up the user email address for GitHub in conjunction with the
+    push token.  It uses "user:email" scope but it's less than useful
+    since private email addresses are still not visible.
+
+    These are all stored in the authenticator's `auth_state`
+    structure, so you'll need to enable `auth_state` and install the
+    Python `cryptography` package to be able to use these.
+
+    You will also need to subclass your spawner to be able to pull
+    these fields out of `auth_state` and use them to provision your
+    Notebook or Lab user.
+
+    """
+
+    use_organizations = False
```

Comment:
These should all probably be traitlets rather than environment variables directly.